swarm/storage/mru/handler.go

```diff
@@ -435,46 +429,48 @@ func (h *Handler) update(ctx context.Context, rootAddr storage.Address, mru *Sig
 
 	rsrc := h.get(rootAddr)
 	if rsrc == nil {
-		return nil, NewError(ErrNotFound, fmt.Sprintf(" object '%s' not in index", rsrc.name))
+		return nil, NewErrorf(ErrNotFound, " object '%s' not in index", rsrc.name)
 	} else if !rsrc.isSynced() {
 		return nil, NewError(ErrNotSynced, " object not in sync")
 	}
 
 	// an update can be only one chunk long; data length less header and signature data
-	if int64(len(mru.data)) > maxUpdateDataLength {
-		return nil, NewError(ErrDataOverflow, fmt.Sprintf("Data overflow: %d / %d bytes", len(mru.data), maxUpdateDataLength))
+	if int64(len(r.data)) > maxUpdateDataLength {
+		return nil, NewErrorf(ErrDataOverflow, "Data overflow: %d / %d bytes", len(r.data), maxUpdateDataLength)
 	}
 
-	if rsrc.period == mru.period {
-		if mru.version != rsrc.version+1 {
-			return nil, NewError(ErrInvalidValue, fmt.Sprintf("Invalid version for this period. Expected version=%d", rsrc.version+1))
+	// Check that the update is consecutive, i.e., that the proposed update
+	// occurs exactly after the latest one we know.
```

Comment:
@jpeletier: After much thought I finally removed this check in the last version in #704 
@nolash: WHY ON EARTH WOULD YOU DO THAT???

Checking that the supplied version/period in a MRU update is immediately after the last known one requires a synced resource, which in turn requires a lookup just to make sure, which is expensive.

If the user is playing nice, he has obtained the MRU update request template by calling `handler.NewUpdateRequest` which already contains the correct `version`, `period` as far as the node knows.
If the user is not playing nice and alters `version`, `period`, well, he will be free to shoot themselves in the foot. The only consequence of this is that their resource won't be found or they could be forking the truth which would be punishable by anyone who finds signatures for the same `version`, `period` with different content. This only affects the resource owner.

Additionally, the evil user, that is, that user who insists on shooting himself/herself on the foot could also build their own node without this check, so we really can't help it.

Another interesting point is that I can now obtain a MRU template from node A, but submit the signed MRU to node B. This second node B does not have to sync the resource. It can just verify the signatures quickly and put the chunk on the network.

Reply:
I don't normally shout ;)

Reply:
Right. Just to confirm then that this lacking check won't allow someone to do the wrong thing _inadvertently_?

Reply:
You won't be able to do the wrong thing inadvertently. If you use the `Request` you are provided, you can't even change `period`, `version` because these are readonly. So you really have to hack a bit to get around this.

Someone could, though, not use `client.go` and post a handcrafted JSON with the wrong `period`, `version`, but again, checking for this kind of thing would be expensive, penalizing most of the people who don't mess with this.

In the docs, I'll make clear not to mess with `period`, `version` when invoking `/bzz-resource:/` directly, or "mess with this at your own risk" kind of warning.

Reply:
i.e., don't mess with the provided `period`, `version` unless you know what you're doing

Reply:
> In the docs, I'll make clear not to mess with period, version when invoking /bzz-resource:/ directly

cool